.github/workflows/samcli-integration-test.yaml

```diff
@@ -0,0 +1,309 @@
+          fi
+
+      - name: Build and initialize Finch
+        run: |
+          make
+          ./bin/finch vm init || ./bin/finch vm start
+          if ! ./bin/finch vm status | grep -q "Running"; then
+            echo "Error: Finch VM is not running"
+            exit 1
+          else
+            echo "Finch VM is running successfully"
+          fi
+
+      - name: Run local invoke test
+        timeout-minutes: 75
+        working-directory: aws-sam-cli
+        run: |
+
+          # Run the test, display output
+          python -m pytest tests/integration/local/invoke -k 'not Terraform' -v > test_output.txt 2>&1 || true
+          cat test_output.txt
+
+          # Create a list of expected failing tests
+          cat > expected_failures.txt << 'EOL'
+          test_invoke_with_error_during_image_build
+          test_invoke_with_timeout_set_0_TimeoutFunction
+          test_invoke_with_timeout_set_1_TimeoutFunctionWithParameter
+          test_invoke_with_timeout_set_2_TimeoutFunctionWithStringParameter
+          test_building_new_rapid_image_removes_old_rapid_images
+          test_invoke_returns_expected_results_from_git_function
+          test_invoke_returns_expected_results_from_git_function_with_parameters
+          EOL
+
+          # Extract actual failing tests
+          grep "FAILED" test_output.txt | sed 's/.*::\(test_[a-zA-Z0-9_]*\).*/\1/' > actual_failures.txt
+
+          # Find unexpected failures (failures not in the expected list)
+          UNEXPECTED_FAILURES=$(grep -v -f expected_failures.txt actual_failures.txt || true)
+
+          # Check if there are any unexpected failures
+          if [ -n "$UNEXPECTED_FAILURES" ]; then
+            echo "Unexpected test failures found:"
+            echo "$UNEXPECTED_FAILURES"
+            echo "Invoke test failed due to unexpected test failures"
+            exit 1
+          else
+            echo "All test failures were expected. Invoke test passed!"
+          fi
+
+      - name: Run start-lambda test
+        timeout-minutes: 30
+        working-directory: aws-sam-cli
+        run: |
+
+          # Run the tests
+          python -m pytest tests/integration/local/start_lambda -k 'not Terraform' -v > start_lambda_output.txt 2>&1 || true
+          cat start_lambda_output.txt
+
+          # Check if any tests failed
+          if grep -q "FAILED" start_lambda_output.txt; then
+            echo "Some start-lambda tests failed!"
+            exit 1
+          else
+            echo "All start-lambda tests passed!"
+          fi
+
+
+      - name: Run start-api test
+        timeout-minutes: 75
+        working-directory: aws-sam-cli
+        run: |
+
+          # Increase file limit to prevent "too many files open" error
+          ulimit -n 4096 || true
+
+          # Run the tests
+          python -m pytest tests/integration/local/start_api -k 'not Terraform' -v > start_api_output.txt 2>&1 || true
```

Comment:
I think you don't need the `|| true` at the end here (and in other places where you do 2>&1) because stdout and stderr will always redirect here